### PR TITLE
[Navigation]: Add animations when changing between active nav items

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
I've used `anchor-positioning` for the animations, so they work without any JS. Because its a fairly new API I've left in a fallback with no animations.

### CODE REVIEW

File: examples/plain-html/index.html [hunk 1]
```diff
@@ -108,6 +108,14 @@
     const segmentedControlToggle = segmentedControlExample.querySelector('leo-toggle');
     const segmentedControl = segmentedControlExample.querySelector('leo-segmentedcontrol');
 
+    for (const navItem of document.querySelectorAll('leo-navigationitem')) {
+      navItem.addEventListener('click', e => {
+        const href = navItem.getAttribute('href')
+        if (!href) return
+        window.location = new URL(href, window.location)
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply examples/plain-html/index.html:115]: Nope this is in an example project only used internally - it's just being used to construct URLs

**Comment by fallaciousreasoning** [comment]
@aguscruiz said my change to the highlight on nested menus should be fine :smile:

**Comment by AlanBreck** [comment]
> @aguscruiz said my change to the highlight on nested menus should be fine 😄

@fallaciousreasoning, are you referring to this?

Screenshot 2024-10-15 at 6 35 39 PM

File: src/components/navigation/navigation.svelte [hunk 2]
```diff
@@ -16,11 +14,44 @@
...
+      border-top-right-radius: var(--leo-radius-xs);
+      border-bottom-right-radius: var(--leo-radius-xs);
+      background: var(--leo-color-text-interactive);
+      position: absolute;
+      left: anchor(left);
+      top: calc(anchor(top) + var(--anchor-padding));
+      bottom: calc(anchor(bottom) + var(--anchor-padding));
+      z-index: 1;
+    }
+  }
```
Comment (AlanBreck) [inline src/components/navigation/navigation.svelte:49]: Feels a bit odd to have one bit of styling here, and the fallback in another file. Is that avoidable?

Comment (fallaciousreasoning) [reply src/components/navigation/navigation.svelte:49]: resolved

**Review by AlanBreck — APPROVED** [review]
Looks good!

File: src/components/navigation/navigation.svelte [hunk 3]
```diff
@@ -16,11 +14,61 @@
...
 </div>
 
 <style lang="scss">
+  :host, .leo-navigation {
+    --anchor-padding: var(--leo-spacing-m);
+    position: relative;
+  }
+
+  :global(leo-navigation::before), .active-indicator {
+    // Anchor positioned active indicator - we have a fallback in navigationItem
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply src/components/navigation/navigation.svelte]: Removed

**Comment by fallaciousreasoning** [comment]
@AlanBreck I'm going to merge this as-is, but I'll have a look at using the `viewTransitions` api when I get a chance

**Comment by kdenhartog** [comment]
Took a look at this and it's not an issue see here: https://github.com/brave/leo/pull/854#discussion_r1783792844

removing the label

**Pull request merged**